lib/rpc/ProviderHealthStateDynamoDbRepository.ts

```diff
@@ -0,0 +1,44 @@
+import { ProviderHealthStateRepository } from './ProviderHealthStateRepository'
+import Logger from 'bunyan'
+import { DynamoDB } from 'aws-sdk'
+import { DocumentClient } from 'aws-sdk/clients/dynamodb'
+import { ProviderHealthState } from './ProviderHealthState'
+
+export class ProviderHealthStateDynamoDbRepository implements ProviderHealthStateRepository {
+  private ddbClient: DynamoDB.DocumentClient
+
+  constructor(private dbTableName: string, private log: Logger) {
+    this.ddbClient = new DynamoDB.DocumentClient()
+  }
+
+  async read(providerId: string): Promise<ProviderHealthState | null> {
```

Comment:
it looks like the provider health state is for all chain : should we handle the case where a provider is good at chain A but not for chain B?

Reply:
No, every `Chain-Provider` pair has a health state.

The DB table looks like this:

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/n7CRFgTfi6wAndDXekPd/718029af-7234-41b0-a59a-58e0e098b61f.png)

Reply:
I see